app/tests/spec/models/auth_brokers/oauth.js

```diff
@@ -260,6 +260,31 @@ function (chai, sinon, Session, p, OAuthClient, Assertion, AuthErrors,
         assert.equal(transformed, '/oauth/signin');
       });
     });
+
+    describe('shouldPromptForPermissions', function () {
+      it('should not prompt when relier is trusted', function () {
+        sinon.stub(relier, 'isTrusted', function () {
+          return true;
+        });
+        assert.isFalse(broker.shouldPromptForPermissions(user.initAccount()));
+        assert.isTrue(relier.isTrusted.called);
+      });
+
+      it('should prompt when relier is untrusted and needs scopes', function () {
+        var account = user.initAccount();
+        sinon.stub(relier, 'isTrusted', function () {
+          return false;
+        });
+        sinon.stub(account, 'hasGrantedPermissions', function () {
+          return false;
```

Comment:
Can you add a test case for `true`? I am seeing odd behavior in that I am being re-asked for permissions on 321done after I have already granted permissions. Is that the desired behavior? It feels a bit odd.